addons/binding/org.openhab.binding.oceanic/src/main/java/org/openhab/binding/oceanic/handler/OceanicThingHandler.java

```diff
@@ -0,0 +1,256 @@
+                                    Thread.sleep(GRACE_PERIOD);
+                                } catch (InterruptedException e) {
+                                    logger.error("An exception occurred while putting the thread to sleep: {}",
+                                            e.getMessage());
+                                }
+                                String response = requestResponse(selector.name());
+
+                                if (response != "") {
+                                    if (selector.isProperty()) {
+                                        logger.debug("Updating the property '{}' with value '{}'", selector.toString(),
+                                                selector.convertValue(response));
+                                        Map<String, String> properties = editProperties();
+                                        properties.put(selector.toString(), selector.convertValue(response));
+                                        updateProperties(properties);
+                                    } else {
+                                        State value = createStateForType(selector, response);
+                                        updateState(theChannelUID, value);
+                                    }
+                                } else {
+                                    logger.warn("Received an empty answer for '{}'", selector.name());
+                                }
+                            }
+                        }
+                    }
+                }
+            } catch (Exception e) {
+                logger.error("An exception occurred while polling the Oceanic Water Softener: '{}'", e.getMessage());
+            }
+        }
+    };
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+
+        if (getThing().getStatus() == ThingStatus.ONLINE) {
+
+            if (command instanceof RefreshType) {
+                // Placeholder for later refinement
+
+                String channelID = channelUID.getId();
+                OceanicChannelSelector selector = OceanicChannelSelector.getValueSelector(channelID,
+                        OceanicChannelSelector.ValueSelectorType.GET);
+
+                if (selector != null) {
+                    try {
+                        Thread.sleep(GRACE_PERIOD);
+                    } catch (InterruptedException e) {
+                        logger.error("An exception occurred while putting the thread to sleep: {}", e.getMessage());
+                    }
+                    String response = requestResponse(selector.name());
+
+                    if (response != "") {
+                        if (selector.isProperty()) {
+                            logger.debug("Updating the property '{}' with value '{}'", selector.toString(),
+                                    selector.convertValue(response));
+                            Map<String, String> properties = editProperties();
```

Comment:
As the properties will never change, maybe you could add a flag in your handler and only execute this code once?